src/make_zero.jl

```diff
@@ -117,11 +117,10 @@ end
         return seen[prev]
     end
     prev2 = prev.contents
-    res = Core.Box()
-    seen[prev] = res
-    res.contents = Base.Ref(
+    res = Core.Box(
         EnzymeCore.make_zero(Core.Typeof(prev2), seen, prev2, Val(copy_if_inactive)),
     )
+    seen[prev] = res
```

Comment:
this ordering feels weird, what if a box is in a recursive datastructure?

Reply:
Hm, I don't see what's wrong? The first time you encounter this `prev`, wherever it lives in the datastructure, you construct the corresponding zeroed value, and then you store it in the IdDict such that if the same identical `prev` is encountered again, `make_zero` will return the same identical `res`

Reply:
Oh, I see what you mean now, self-referential data structures hadn't even crossed my mind while working on this. Will make sure to add tests for that to whichever PR gets merged.

Reply:
The actual point of the change here was to get rid of the spurious Ref. The reordering was collateral.